Dockerfile

```diff
@@ -1,14 +1,21 @@
-FROM ubuntu:latest
-RUN apt update
-# Avoid prompt for locale.
-RUN DEBIAN_FRONTEND="noninteractive" apt install -y tzdata
-RUN apt install -y git netcat wget
-RUN apt install -y openjdk-11-jdk maven
-# Used for waiting for services before starting Olog.
-RUN apt install -y wait-for-it
+# Use Maven image to execute build.
+FROM maven:3.6.3-openjdk-11 AS maven-build
 RUN mkdir olog-es
 WORKDIR /olog-es
 COPY . .
 RUN mvn clean install -DskipTests=true -Pdeployable-jar
+
+# Use smaller openjdk image for running.
+FROM openjdk:11
+# apt clean is run automatically in debian-based images.
+RUN apt update && apt install -y wait-for-it
+# Run commands as user 'olog'
+RUN useradd -ms /bin/bash olog
+# Use previous maven-build image.
+COPY --from=maven-build /olog-es/target /olog-target
+RUN chown olog:olog /olog-target
+USER olog
+WORKDIR /olog-target
+EXPOSE 8080
 EXPOSE 8181
-CMD java -jar target/olog-es*.jar
+CMD java -jar olog-es*.jar
```

Comment:
You should use the exec form of [CMD](https://docs.docker.com/engine/reference/builder/#cmd) whenever possible. You are currently using the shell form, which you don't need. Using the exec form also has the advantage of respecting SIGINT signals (Ctrl+C) to interrupt the container.
And while your fixing that, you may as well add a line break at the end of the file to get rid of that warning.